fluss-client/src/main/java/org/apache/fluss/client/write/WriteRecord.java

```diff
@@ -127,6 +127,25 @@ public static WriteRecord forArrowAppend(
                 estimatedSizeInBytes);
     }
 
+    /** Creates a write record for append operation for Compacted format. */
+    public static WriteRecord forCompactedAppend(
+            TableInfo tableInfo,
+            PhysicalTablePath tablePath,
+            CompactedRow row,
+            @Nullable byte[] bucketKey) {
+        checkNotNull(row);
+        int estimatedSizeInBytes = CompactedLogRecord.sizeOf(row) + RECORD_BATCH_HEADER_SIZE;
```

Comment:
This `RECORD_BATCH_HEADER_SIZE` is for kv record batch, we should use `LogRecordBatchFormat.recordBatchHeaderSize(CURRENT_LOG_MAGIC_VALUE)`